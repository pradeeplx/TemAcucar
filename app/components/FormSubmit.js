import React from 'react'
import BottomView from "./BottomView"
import BottomButton from "./BottomButton"

export default FormSubmit = (props) => (
  <BottomView>
    <BottomButton
      isDisabled={!props.dirty || !props.valid || props.submitting}
      onPress={props.handleSubmit(props.onSubmit)}
      {...props}
    >
      {props.children}
    </BottomButton>
  </BottomView>
)
