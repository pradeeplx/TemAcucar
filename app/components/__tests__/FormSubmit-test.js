import 'react-native'
import React from 'react'
import FormSubmit from '../FormSubmit'
import renderer from 'react-test-renderer'
import { identity } from 'lodash'

it('renders correctly', () => {
  const tree = renderer.create(
    <FormSubmit handleSubmit={identity}>Foo</FormSubmit>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('has enabled button', () => {
  const tree = renderer.create(
    <FormSubmit handleSubmit={identity} dirty valid>Foo</FormSubmit>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
