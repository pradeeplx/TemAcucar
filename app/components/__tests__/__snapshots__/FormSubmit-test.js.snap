exports[`test has enabled button 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
      },
      undefined,
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#FF7F85",
        "borderRadius": 0,
        "borderWidth": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": 105,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 0,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
            undefined,
          ],
        ]
      }>
      Foo
    </Text>
  </View>
</View>
`;

exports[`test renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
      },
      undefined,
    ]
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 44,
          "justifyContent": "center",
          "marginBottom": 10,
        },
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#FF7F85",
            "borderRadius": 0,
            "borderWidth": 0,
            "flex": 1,
            "height": 105,
            "marginBottom": 0,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "width": 0,
          },
          undefined,
        ],
        Array [
          Object {
            "opacity": 0.6,
          },
          undefined,
        ],
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
            undefined,
          ],
        ]
      }>
      Foo
    </Text>
  </View>
</View>
`;
